_clients/ruby.md

```diff
@@ -0,0 +1,528 @@
+
+You can verify the index's mappings using the `get_mapping` method:
+
+```ruby
+response = client.indices.get_mapping(index: 'students')
+```
+
+If you know the mapping of your documents in advance and want to avoid mapping errors (for example, misspellings of a field name), you can set the `dynamic` parameter to `strict`:
+
+```ruby
+client.indices.put_mapping(
+    index: 'students', 
+    body: {
+        dynamic: 'strict',
+        properties: {
+            first_name: { type: 'keyword' },
+            last_name: { type: 'keyword' },
+            gpa: { type: 'float'},
+            grad_year: { type: 'integer'}
+        }  
+    }
+)
+```
+
+With strict mapping, you can index a document where a field is missing, but you cannot index a document with a new field. For example, indexing the following document with a misspelled `grad_yea` field fails:
+
+```ruby
+document = {
+    first_name: 'Connor',
+    last_name: 'James',
+    gpa: 3.93,
+    grad_yea: 2021
+}
+  
+client.index(
+    index: 'students',
+    body: document,
+    id: 100,
+    refresh: true
+)
+```
+
+OpenSearch returns a mapping error:
+
+```bash
+{"error":{"root_cause":[{"type":"strict_dynamic_mapping_exception","reason":"mapping set to strict, dynamic introduction of [grad_yea] within [_doc] is not allowed"}],"type":"strict_dynamic_mapping_exception","reason":"mapping set to strict, dynamic introduction of [grad_yea] within [_doc] is not allowed"},"status":400}
+```
+
+## Indexing one document
+
+To index one document, use the `index` method:
+
+```ruby
+document = {
+    first_name: 'Connor',
+    last_name: 'James',
+    gpa: 3.93,
+    grad_year: 2021
+}
+  
+client.index(
+    index: 'students',
+    body: document,
+    id: 100,
+    refresh: true
+)
+```
+
+## Updating a document
+
+To update a document, use the `update` method:
+
+```ruby
+client.update(index: 'students', 
+              id: 100, 
+              body: { doc: { gpa: 3.25 } }, 
+              refresh: true)
+```
+
+## Deleting a document
+
+To delete a document, use the `delete` method:
+
+```ruby
+client.delete(
+    index: 'students',
+    id: 100,
+    refresh: true
+)
+```
+
+## Bulk operations
+
+You can perform several operations at the same time by using the `bulk` method. The operations may be of the same type or of different types.
+
+You can index multiple documents as follows:
+
+```ruby
+actions = [
+    { index: { _index: 'students', _id: '200' } },
+    { first_name: 'James', last_name: 'Rodriguez', gpa: 3.91, grad_year: 2019 },
+    { index: { _index: 'students', _id: '300' } },
+    { first_name: 'Nikki', last_name: 'Wolf', gpa: 3.87, grad_year: 2020 }
+]
+client.bulk(body: actions, refresh: true)
+```
+
+You can delete multiple documents as follows:
```

Comment:
Same comment re: "as follows". "As follows" is fine in sentences like "The output is as follows" because we're stating what is shown in the example in an exact and unambiguous way, but when we're referring to a way of doing something, we should be more explicit/clear in our language by explaining what is done/performed in the example that follows.